qiskit_experiments/data_processing/data_action.py

```diff
@@ -30,46 +30,74 @@ def __init__(self, validate: bool = True):
         self._validate = validate
 
     @abstractmethod
-    def _process(self, datum: Any) -> Any:
+    def _process(self, datum: Any, error: Optional[Any] = None, **options) -> Tuple[Any, Any]:
         """
         Applies the data processing step to the datum.
 
         Args:
             datum: A single item of data which will be processed.
+            error: An optional error estimation on the datum that can be further propagated.
+            options: Keyword arguments passed through the data processor at run-time.
 
         Returns:
-            processed data: The data that has been processed.
+            processed data: The data that has been processed along with the propagated error.
         """
 
     @abstractmethod
-    def _format_data(self, datum: Any) -> Any:
-        """
-        Check that the given data has the correct structure. This method may
+    def _format_data(self, datum: Any, error: Optional[Any] = None) -> Tuple[Any, Any]:
+        """Format and validate the input.
+
+        Check that the given data and error has the correct structure. This method may
         additionally change the data type, e.g. converting a list to a numpy array.
 
         Args:
             datum: The data instance to check and format.
+            error: An optional error estimation on the datum to check and format.
 
         Returns:
-            datum: The data that was checked.
+            datum, error: The formatted datum and its optional error.
 
         Raises:
-            DataProcessorError: If the data does not have the proper format.
+            DataProcessorError: If either the data or the error do not have the proper format.
         """
 
-    def __call__(self, data: Any) -> Any:
-        """
-        Call the data action of this node on the data.
+    def __call__(self, data: Any, error: Optional[Any] = None, **options) -> Tuple[Any, Any]:
+        """Call the data action of this node on the data and propagate the error.
 
         Args:
             data: The data to process. The action nodes in the data processor will
                 raise errors if the data does not have the appropriate format.
+            error: An optional error estimation on the datum that can be further processed.
+            options: Keyword arguments passed through the data processor at run-time.
 
         Returns:
-            processed data: The data processed by self.
+            processed data: The data processed by self as a tuple of processed datum and
+                optionally the propagated error estimate.
         """
-        return self._process(self._format_data(data))
+        return self._process(*self._format_data(data, error), **options)
 
     def __repr__(self):
         """String representation of the node."""
         return f"{self.__class__.__name__}(validate={self._validate})"
+
+    @property
+    def is_trained(self) -> bool:
```

Comment:
This doesn't really seem like it should be a property of the base class, since it is only used in specific nodes. Training should probably be part of those subclasses that require it.

Reply:
As discussed, I introduced a `TrainableDataAction` see here: 74e928b